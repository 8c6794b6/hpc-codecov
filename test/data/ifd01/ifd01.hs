module Main where

pre :: R -> IO ()
pre (R f1 f2)
  | f1 = putStrLn "=== ifd01.hs: f1 ==="
  | f2 = putStrLn "=== ifd01.hs: f2 ==="
  | otherwise = putStrLn "=== ifd01.hs ==="

data R = R
  { field1 :: Bool
  , field2 :: Bool
  } deriving (Show)

post :: R -> IO ()
post (R f1 f2)
  | f1 = putStrLn "=== the end: f1 ==="
  | f2 = putStrLn "=== the end: f2 ==="
  | otherwise = putStrLn "=== the end ==="

main :: IO ()
main = do
  pre $ R True False
  print $ R True False
  post $ R False True
