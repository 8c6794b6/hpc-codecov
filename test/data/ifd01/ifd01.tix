Tix [ TixModule "Main" 1766259647 56 [0,1,1,0,0,1,0,1,1,1,0,0,0,0,0,1,1,0,1,1,1,0,0,0,0,0,0,0,0,0,0,1,1,1,0,1,1,1,1,1,1,1,1,1,1,1,1,1,1,0,0,0,0,1,1,0]]
