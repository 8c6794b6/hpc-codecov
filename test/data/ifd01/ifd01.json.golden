{"coverage":{"test/data/ifd01/ifd01.hs":{"4":1,"5":"1/2","6":0,"7":0,"15":1,"16":"1/2","17":"1/2","18":0,"21":1,"22":"1/2","23":1,"24":1}}}
